packages/plugin-lighthouse/src/lib/utils.unit.test.ts

```diff
@@ -54,3 +59,265 @@ describe('validateOnlyAudits', () => {
+          type: 'treemap-data',
+          nodes: [],
+        },
+      },
+    });
+
+    expect(outputs[0]?.details?.issues[0]).toEqual(
+      expect.objectContaining({
+        message: 'Parsing details from type treemap-data is not implemented.',
+        severity: 'info',
+      }),
+    );
+  });
+
+  it('should parse valid lhr details criticalrequestchain', () => {
+    const outputs = toAuditOutputs({
+      audit: {
+        scoreDisplayMode: 'binary',
+        title: 'audit',
+        description: 'audit',
+        score: null,
+        id: 'audit',
+        details: {
+          type: 'criticalrequestchain',
+          chains: {
+            EED301D300C9A7B634A444E0C6019FC1: {
+              request: {
+                url: 'https://example.com/',
+                startTime: 106_245.050_727,
+                endTime: 106_245.559_225,
+                responseReceivedTime: 106_245.559_001,
+                transferSize: 849,
+              },
+            },
+          },
+          longestChain: {
+            duration: 508.498_000_010_848_05,
+            length: 1,
+            transferSize: 849,
+          },
+        },
+      },
+    });
+
+    expect(outputs[0]?.details?.issues[0]).toEqual(
+      expect.objectContaining({
+        message:
+          'Parsing details from type criticalrequestchain is not implemented.',
+        severity: 'info',
+      }),
+    );
+  });
+});
+
+describe('tableToDetails', () => {
+  it('should parse empty lhr details table', () => {
+    expect(
+      tableToDetails({
+        type: 'table',
+        headings: [],
+        items: [],
+      } satisfies Details.Table),
+    ).toStrictEqual({
+      issues: [
+        {
+          message: 'no data present',
+          severity: 'info',
+        } satisfies Issue,
+      ],
+    });
+  });
+
+  it('should parse lhr details table with data', () => {
+    expect(
+      tableToDetails({
+        type: 'table',
+        headings: [
+          {
+            key: 'sourceLocation',
+            valueType: 'source-location',
+            label: 'Source',
+          },
+          {
+            key: 'description',
+            valueType: 'code',
+            label: 'Description',
+          },
+        ],
+        items: [
+          {
+            source: 'network',
+            description:
+              'Failed to load resource: the server responded with a status of 404 ()',
+            sourceLocation: {
+              type: 'source-location',
+              url: 'https://example.com/favicon.ico',
+              urlProvider: 'network',
+              line: 0,
+              column: 0,
+            },
+          },
+        ],
+      } satisfies Details.Table),
+    ).toStrictEqual({
+      issues: [
+        {
+          message: 'sourceLocation, description',
+          severity: 'info',
+        } satisfies Issue,
+      ],
+    });
+  });
+});
+
+describe('opportunityToDetails', () => {
+  it('should parse valid lhr details', () => {
```

Comment:
I like the description distinction you did in the table that distinguishes valid/invalid and empty/non-empty.

```suggestion
  it('should parse a valid non-empty lhr opportunity type', () => {
```